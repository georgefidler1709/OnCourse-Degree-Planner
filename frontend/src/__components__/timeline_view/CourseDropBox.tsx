import React from 'react';
import styled from 'styled-components';
import { Droppable } from 'react-beautiful-dnd';
import { Course as ApiCourse } from "../../Api";
import Course from './Course';

const Container = styled.div`
  margin: 8px;
  border: 1px solid lightgrey;
  border-radius: 2px;
  width: 220px;

  display: flex;
  flex-direction: column;
`;
const Title = styled.h5`
  padding: 8px;
`;
const CourseList = styled.div`
  padding: 8px;
  flex-grow: 1;
  min-height: 50px;
`;

interface DropBoxProps {
  type: string;
  add_course: ApiCourse;
  remove_course: (id: string) => void;
}

function CourseDropBox(props: DropBoxProps) {

  if (props.add_course !== undefined) {
    // make a course and put it in the box
    return (
      <Container>
        <Title>{props.type}</Title>
        <Droppable droppableId={props.type}>
          {provided => (
            <CourseList innerRef={provided.innerRef} {...provided.droppableProps}>
              <Course
                {...props.add_course}
                key={props.add_course.code}
                index={0}
                removeCourse={props.remove_course}
              />
            </CourseList>
          )}
        </Droppable>
      </Container>
    );
  } else {
    // placeholder box
    return (
      <Container>
        <Title>{props.type}</Title>
        <Droppable droppableId={props.type}>
          {provided => (
            <CourseList innerRef={provided.innerRef} {...provided.droppableProps}>
              {provided.placeholder}
            </CourseList>
          )}
        </Droppable>
      </Container>
    );

  }

}

export default CourseDropBox
