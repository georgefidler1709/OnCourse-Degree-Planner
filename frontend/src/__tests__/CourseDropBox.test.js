import React from 'react'
import { shallow, mount } from 'enzyme';
import CourseDropBox from '../__components__/timeline_view/CourseDropBox';
import { DragDropContext, Droppable } from 'react-beautiful-dnd';

console.error = jest.fn();

describe('Render degree planning timeline view', () => {
  it('renders correctly', () => {
    const wrapper = mount(
        <DragDropContext>
        {
         <Droppable droppableId="droppable">
         {(provided, snapshot) => (
           <div 
             innerRef={provided.innerRef} 
             {...provided.droppableProps}
           >
             {
                <CourseDropBox type="test"/>
             }
             {provided.placeholder}
           </div>
         )}
       </Droppable>
        }
        </DragDropContext>
        );
        expect(wrapper).toMatchSnapshot();
        
        wrapper.unmount();
  })
});
