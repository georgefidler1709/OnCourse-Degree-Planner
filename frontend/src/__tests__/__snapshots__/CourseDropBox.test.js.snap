// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render degree planning timeline view renders correctly 1`] = `ReactWrapper {}`;
